app/models/item.rb

```diff
@@ -1,26 +1,31 @@
 class Item < ApplicationRecord
   acts_as_paranoid
 
+  belongs_to :user
+  belongs_to :list
+
   has_many :links, dependent: :delete_all
   has_many :notes, dependent: :delete_all
 
-  has_and_belongs_to_many :labels
   has_and_belongs_to_many :tags
 
-  belongs_to :user
+  # Deprecated. This statement and the items_lists table will be removed after
+  # the next deployment.
+  has_and_belongs_to_many :lists
 
   mount_uploader :image, ImageUploader
 
+  validates :list, presence: true
   validates :name, presence: true, uniqueness: { scope: [:user_id] }
 
   after_create :scrape_in_background, unless: -> { Rails.env.test? || scraped? }
 
-  def self.with_labels(label_ids)
-    joins(:items_labels).where('items_labels.label_id IN (?)', label_ids).distinct
+  def self.in_lists(list_ids)
```

Comment:
i think it would help to add a comment what type of input this method expects: 

```ruby
# @param list_ids [Array<Integer>]
def self.in_lists(list_ids)
...
```